### PR TITLE
Deprecate the Stream and Genlex stdlib modules

### PR DESCRIPTION
**Pull request by xavierleroy**
The core OCaml development team decided to remove the Stream and Genlex modules from the standard library, making them available in a separate library [camlp-streams](https://github.com/ocaml/camlp-streams) instead.

(Rationale: these two modules are tightly connected with Camlp4/Camlp5 stream parsers and printers, but not really usable by themselves.   With minor changes to Camlp4 and Camlp5, the separate library camlp-streams can be automatically pulled in programs that use Camlp4/Camlp5 parsers.)

To ease the transition, this PR adds a "deprecated" alert to the Stream and Genlex stdlib modules.

If the PR is merged soon, the deprecation will become active in OCaml 4.14, to be released early 2022.  Removal of Stream and Genlex could take place one year after.

### CODE REVIEW

File: stdlib/stdlib.mli
Comment (nojb) [inline stdlib/stdlib.mli]:
```suggestion
[@@ocaml.deprecated "Use the camlp-streams library instead."]
```

**Review by nojb — APPROVED** [review]
LGTM. A `Changes` entry is in order I believe.

File: stdlib/stdlib.mli
Comment (xavierleroy) [reply stdlib/stdlib.mli]: Well spotted.  I reworded both warnings.

**Comment by xavierleroy** [comment]
Let's wait a bit before merging this PR, while the camlp-streams package makes its way through OPAM, and to give the Camlp4 and Camlp5 maintainers an opportunity to comment.

**Comment by gasche** [comment]
Note: this PR broke the dune build of the compiler (which I'm using to get decent Merlin support), because (at least) genlex.mli mentions Stream and fails to compile with a deprecation error.

**Pull request merged**
